### PR TITLE
[Networking] Chore: refactors the lifecycle management of alsp manager

### PR DESCRIPTION
**Pull request by yhassanzadeh13**
This Pull Request is a continuation of https://github.com/onflow/flow-go/pull/4361, which further enhances the code by removing component behavior from the `ConduitFactory` interface. Moreover, the lifecycle management of the `MisbehaviorReportManager` has been repositioned to be under the parent `Network` interface.

### CODE REVIEW

File: network/p2p/conduit/conduit.go [hunk 1]
```diff
@@ -41,36 +28,8 @@ func WithMisbehaviorManager(misbehaviorManager network.MisbehaviorReportManager)
...
-			}
-
-			<-ctx.Done()
-			<-d.misbehaviorManager.Done()
-		}).Build()
-
-	d.Component = cm
-
-	return d, nil
+func NewDefaultConduitFactory() *DefaultConduitFactory {
```
Comment (peterargue) [inline network/p2p/conduit/conduit.go:31]: don't forget to update the godocs

**Review by peterargue — APPROVED** [review]
thanks!

**Review by kc1116 — APPROVED** [review]

File: network/p2p/conduit/conduit.go (hunk 1 above)
Comment (yhassanzadeh13) [reply network/p2p/conduit/conduit.go:27]: https://github.com/onflow/flow-go/pull/4376/commits/5fa7257ac5d2206818ecbc0cb1a839512c5b46e9

**Comment by yhassanzadeh13** [comment]
bors merge

**Comment by yhassanzadeh13** [comment]
bors merge

**Pull request merged**